test/MUIDataTable.test.js

```diff
@@ -429,6 +430,24 @@ describe('<MUIDataTable />', function() {
     assert.deepEqual(state.filterList, [['Joe James'], [], [], [], []]);
   });
 
+  it('should apply columns prop change for filterList', () => {
+    const mountShallowWrapper = mount(shallow(<MUIDataTable columns={columns} data={data} />).get(0));
+    const instance = mountShallowWrapper.instance();
+    instance.initializeTable(mountShallowWrapper.props());
+    // now use updated columns props
+    const newColumns = cloneDeep(columns);
+    newColumns[0].options.filterList = ['Joe James'];
+    mountShallowWrapper.setProps({ columns: newColumns });
+    mountShallowWrapper.update();
+    // mimic componentDidUpdate(prevProps) {
+    // if (this.props.data !== prevProps.data || this.props.columns !== prevProps.columns) {
+    instance.setTableData(mountShallowWrapper.props(), 1 /* instance.TABLE_LOAD.INITIAL */);
+
+    const updatedState = mountShallowWrapper.state();
+    const { columns: updatedColumns } = updatedState;
```

Comment:
This line doesn't look like it's doing anything in the test, so I think we can remove it.

Reply:
You're right. Thanks.